force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -71,8 +89,18 @@ public with sharing class FlowControlLoopsCollections {
      * @return A string indicating whether the number is "Positive", "Negative", or "Zero", or null if the number is null.
      */
     public static String checkNumber(Integer a) {
-        return null; // Replace null with the variable you used to store the result
-    }
+        if (a < 0 ) {
+            return 'Negative';
+        } else if (a > 0) {
+            return 'Positive';
+        } else if (a == 0) {
+            return 'Zero';
+        } else {
+            return null;
+            }
+        }
+         return checkNumber; // Replace null with the variable you used to store the result
+    
```

Comment:
The conditional logic looks correct. But this could should not compile. It appears that you are missing your closing }

and you are returning the variable with the name checkNumber; But I don't see a declaration or assignment anywhere.